sotodlib/site_pipeline/check_book.py

```diff
@@ -0,0 +1,422 @@
+"""check_book.py
+
+This module contains a class for checking obs/oper Books (for internal
+consistency, proper schema, etc).  It can also be run to create/update
+an ObsFileDb for such Books.
+
+The config file for this should look something like this:
+
+.. code-block:: yaml
+
+    # Database setup
+    obsfile_db: './obsfiledb.sqlite'
+    root_path: '/'
+
+    # Work-arounds
+    stream_file_pattern: 'D_obs_{stream_id}_{index:03d}.g3'
```

Comment:
a heads-up that I'm fixing the D-prefixed file naming in #401 and will propagate the changes to previous books once that's merged.